### PR TITLE
SAA-690 - deallocation end date not updating

### PR DESCRIPTION
**Pull request by gerryallan-moj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/AllocationsService.kt [hunk 1]
```diff
@@ -109,6 +109,8 @@ class AllocationsService(
       }
       if (allocation.endDate == null) {
         allocation.activitySchedule.deallocatePrisonerOn(allocation.prisonerNumber, this, request.reasonCode.toDeallocationReason(), updatedBy)
+      } else {
+        allocation.activitySchedule.deallocatePrisonerOn(allocation.prisonerNumber, this, allocation.plannedDeallocation!!.plannedReason, updatedBy)
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/AllocationsService.kt:113]: Can do with some test coverage please if there isn't any.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/AllocationsService.kt [hunk 2]
```diff
@@ -109,6 +109,8 @@ class AllocationsService(
...
       }
```
Comment (timharrison-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/AllocationsService.kt:114]: Not sure about this change, Gerry. Can we always assume a `plannedDeallocation` exists and assert it is not null when changing the end date?

Comment (gerryallan-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/AllocationsService.kt:114]: if endDate has already been set then there is always a plannedDeallocation

**Review by timharrison-moj — APPROVED** [review]
LGTM

**Pull request merged**